dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,344 @@
+}
+void NumberBox::OnSpinButtonPlacementModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    SetSpinButtonVisualState();
+}
+void NumberBox::OnPlaceholderTextPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    SetPlaceHolderText();
+}
+
+void NumberBox::OnFractionDigitsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    m_formatter.FractionDigits(FractionDigits());
+}
+void NumberBox::OnIntegerDigitsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    m_formatter.IntegerDigits(IntegerDigits());
+}
+void NumberBox::OnSignificantDigitsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    m_formatter.SignificantDigits(SignificantDigits());
+}
+void NumberBox::OnIsDecimalPointAlwaysDisplayedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    m_formatter.IsDecimalPointAlwaysDisplayed(IsDecimalPointAlwaysDisplayed());
+}
+void NumberBox::OnIsZeroSignedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    m_formatter.IsZeroSigned(IsZeroSigned());
+}
+
+void NumberBox::OnRoundingAlgorithmPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnNumberRounderPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnIncrementPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+void NumberBox::OnSignificantDigitPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateRounder();
+}
+
+void NumberBox::OnValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (m_TextBox)
+    {
+       // UpdateTextToValue();
+    }
+}
+
+
+
+// Trigger any validation, rounding, and processing done onLostFocus
+void NumberBox::OnTextBoxLostFocus(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    ValidateInput();
+}
+
+// Performs all validation steps on input given in textbox. Runs on LoseFocus and stepping.
+void NumberBox::ValidateInput()
+{
+    std::string InputAsString = winrt::to_string(m_TextBox.Text());
+    // Handles Empty TextBox Case, current behavior is to set Value to default (0)
+    if (InputAsString == "")
+    {
+        Value(0);
+        m_hasError = false;
+        return;
+    }
+
+    winrt::IReference<double> parsedNum = m_formatter.ParseDouble(m_TextBox.Text());
+    // TODO: Formatter.Format(value)
+
+    if (parsedNum && IsInBounds(parsedNum.Value()) )
+    {
+        SetErrorState(false);
+        Value(parsedNum.Value());
+        UpdateTextToValue();
+    }
+    else
+    {
+        SetErrorState(true);
+    }
+}
+
+// SpinClicks call to decrement or increment, 
+void NumberBox::OnSpinDownClick(winrt::IInspectable const&  sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(false);
```

Comment:
>false [](start = 14, length = 5)

It is hard to tell what this does from the signature. Can you either make helpers like StepValueUp() and StepValueDown() which calls this as written, or annotate the paramter: StepValue(sign = true). if you choose the second, maybe a better name for sign would be good too.